src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -1059,6 +1099,12 @@ sol_http_client_connection_cancel(struct sol_http_client_connection *pending)
 {
     SOL_NULL_CHECK(pending);
 
+    if (pending->in_use) {
+        pending->del_timeout = sol_timeout_add(0, schedule_del, pending);
+        SOL_NULL_CHECK(pending->del_timeout);
```

Comment:
this is not good since in cases where the main loop is finalized/quit, the timer may not run.

Why not use a `delete_me` as you did in other code? Maybe check crypto/message-digest and use the flag with a refcount?